### PR TITLE
Improve stm32 chibios lwipthread alignment

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
Re-Align lwipthread with the latest ChibiOS source.

## Motivation and Context
After looking at SNTP issues, it was apparent that the source was a little different (even after recommiting the code style changes). It mainly involved commented out IPV6 code and SNTP changes.
I have removed the IPV6 changes as they should have been a seperate PR.
I have kept the SNTP changes evn though the code should be re-evaluated.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: targets/ChibiOS/_Lwip/nf_lwipthread.c [hunk 1]
```diff
@@ -341,24 +334,20 @@ static THD_FUNCTION(lwip_thread, p)
...
-#if LWIP_IPV6
-        // FIXME IPV6
-        // LWIP_IPADDR(&ip);
-        // LWIP_GATEWAY(&gateway);
-        // LWIP_NETMASK(&netmask);
-#else
         LWIP_IPADDR(&ip);
         LWIP_GATEWAY(&gateway);
         LWIP_NETMASK(&netmask);
+#if LWIP_DHCP
```
Comment (Ellerbach) [inline targets/ChibiOS/_Lwip/nf_lwipthread.c:340]: can't that be dynamic?

**Review by Ellerbach — APPROVED** [review]
Looks good to me! But why do you have to hard code the type of network?

File: targets/ChibiOS/_Lwip/nf_lwipthread.c (hunk 1 above)
Comment (networkfusion) [reply targets/ChibiOS/_Lwip/nf_lwipthread.c:340]: Perhaps... but I was re-aligning with https://github.com/ChibiOS/ChibiOS/blob/stable_21.11.x/os/various/lwip_bindings/lwipthread.c which uses the above...

Comment (networkfusion) [reply targets/ChibiOS/_Lwip/nf_lwipthread.c:340]: I do indeed think it is weird. But, if you check the changes, this only confirms what is done in the unchanged parts of the file. Perhaps that would be the reason for some other issues with STM32 networking 😅

Comment (networkfusion) [reply targets/ChibiOS/_Lwip/nf_lwipthread.c:340]: For notes: most of the change history is here: https://github.com/nanoframework/nf-interpreter/commits/f648f41ed11399077bdc3e863c924467df4d86de/targets/CMSIS-OS/ChibiOS/Lwip/nf_lwipthread.c?browsing_rename_history=true&new_path=targets/ChibiOS/_Lwip/nf_lwipthread.c&original_branch=main

**Comment by networkfusion** [comment]
Converting this to draft until I can test further.

**Comment by networkfusion** [comment]
Closing this as ChibiOS is needing further rework, which I am unable to provide.

**Pull request closed without merging**